mp/src/game/shared/momentum/run/mom_run_safeguards.cpp

```diff
@@ -130,8 +130,10 @@ bool CRunSafeguard::IsDoublePressSafeguarded()
 {
     if (gpGlobals->curtime > m_flLastTimePressed + mom_run_safeguard_doublepress_maxtime.GetFloat())
     {
-        Warning("You must double press the key to %s while the timer is running!\n"
-                "You can turn this safeguard off in the Gameplay Settings!\n", m_szAction);
+        #ifdef GAME_DLL
+        UTIL_ShowMessage("SAFEGUARD_DOUBLE_PRESS", UTIL_GetLocalPlayer());
+        #endif
```

Comment:
What will surely get annoying is that this will pop up even when the player double presses the key successfully. I suggest clearing the message in when it isn't safeguarded. Here and also in the movement keys safeguard.

Reply:
How do you clear the hud message? `UTIL_ShowMessage("", UTIL_GetLocalPlayer())`?

Reply:
Oh, didn't realize this didnt have a function for that. Perhaps use hud hints instead as they can be cleared this way (by sending an empty string). They can also display the key it's referring to and have localization.
See how it's used for the start mark creation via saveloc create here
https://github.com/momentum-mod/game/blob/a12d083604bf56c9888a5c92822e2b24af262e38/mp/src/game/server/momentum/mom_system_saveloc.cpp#L507
and here
https://github.com/momentum-mod/game/blob/f63af112ebe6ca2e7e9edb6dc04650faa845007a/mp/game/momentum/resource/momentum_english.txt#L177
the command surrounded in `%`'s has the bind for it placed there.

Reply:
I'm unable to get `UTIL_HudHintText()` to interpret %s, it keeps trying to interpret it as a keybind but I'm unsure if its just me sucking at implementing it properly

Reply:
Just need to pass the localization token in. Im thinking the best way is to have a localization token for each of the distinct messages, passed in through the constructor. That includes the 2 modes for each safeguard type, so like 10 localization strings altogether. 

Maybe @Gocnak has a better solution.

Reply:
Ah okay; I had that idea but spent the entirety of last night trying to find a different way using `%s` with the token if that makes sense; just didn't want to write overtly redundant tokens